src/shared/telemetry/telemetryUtils.ts

```diff
@@ -42,15 +57,16 @@ export function registerCommand<T>({
                 throw e
             } finally {
                 const endTime = new Date()
-                const datum = result && result.datum ? result.datum : defaultMetricDatum(command)
+                const datum = result && result.datum ?
+                    result.datum : defaultMetricDatum(telemetryName ? telemetryName.name : command)
                 if (!datum.metadata) {
                     datum.metadata = new Map()
                 }
-                datum.metadata.set('hasException', `${hasException}`)
+                datum.metadata.set('result', hasException ? 'Failed' : 'Succeeded')
                 datum.metadata.set('duration', `${endTime.getTime() - startTime.getTime()}`)
 
                 ext.telemetry.record({
-                    namespace: 'Command',
+                    namespace: telemetryName ? telemetryName.namespace : 'Command',
```

Comment:
1. Consider adding `'Command'` to the enum, or if it should be considered more "private" than the enum, consider defining a constant and commenting its intended use.
2. Is there a test for this and if not is it possible for there to be?

Reply:
I'll take a look into the test. I don't want to add `'Command'` to the enum since that's not a metric we're going to track--this is for backwards compatibility for unported metrics. We'll set this to a different type.

Reply:
✔️